bika/lims/browser/bika_listing.py

```diff
@@ -791,6 +728,99 @@ def _process_request(self):
             else:
                 self.columns[col]['toggle'] = False
 
+    def _set_sorting_criteria(self):
+        """
+        Sets the sorting criteria by resetting the value of sort_on, sort_order
+        and/or manual_sort in accordance with the values set in the request,
+        contentFilter and by default for this list
+        """
+        self.sort_on = self.get_sort_on()
+        self.manual_sort_on = None
+        if 'sort_on' in self.contentFilter:
+            del self.contentFilter['sort_on']
+
+        allowed_indexes = self.get_columns_indexes()
+        allowed_indexes.append('created')
+        if self.sort_on not in allowed_indexes:
+            # The value for sort_on is not an index, we need to force manual
+            # sorting so items will be sorted programmatically with python
+            self.manual_sort_on = self.sort_on
+
+        else:
+            # sort_on is an index, add it into the contentFilter so it can be
+            # used in the advanced query later
+            self.contentFilter['sort_on'] = self.sort_on
+
+        # By default, if sort_on is set, sort the items ASC
+        # Trick to allow 'descending' keyword instead of 'reverse'
+        sort_order = self.request.get(self.form_id + "_sort_order", None)
+        if not sort_order:
+            sort_order = self.contentFilter.get('sort_order', self.sort_order)
+        if sort_order != "ascending":
+            sort_order = "descending"
+        self.contentFilter['sort_order'] = sort_order
+        self.sort_order = sort_order
+
+    def get_sort_on(self):
+        """
+        Returns the value by which this list must be sorted on.
+        Returns the name of the index to be used for sorting by using an
+        advancedQuery if defined in self.columns. Otherwise, returns the name
+        of the column the list must be sorted on.
+        :return: the sort_on index or column name
+        :rtype: str
+        """
+
+        def corrected_sort_value(value):
```

Comment:
I had to check the same (if the value is set in columns and/or in indexes) for each `sort_on` possible sources (`self.request`, `self.contentFilter` and `self.sort_on`), so instead of triplicating the same code or placing it into the loop (see L808), I though this way would be better.